apps/system/js/audio_channel_manager.js

```diff
@@ -0,0 +1,370 @@
+        app.browser.element.allowedAudioChannels.forEach((audioChannel) => {
+          var id = app.getAudioChannel(audioChannel.name).instanceID;
+          this._activeAudioChannels.delete(id);
+          var element = this._interruptedAudioChannels.get(id);
+          if (element) {
+            this._interruptedAudioChannels.delete(id);
+            var index = element.backgroundResumeIndex;
+            index != null &&
+              this._audioChannelsResumeWhenOtherEnds.splice(index, 1);
+          }
+        });
+      }
+    },
+
+    /**
+     * Observer the value of vibration.enabled settings.
+     *
+     * @param {Boolean} value The value of the settings.
+     */
+    '_observe_vibration.enabled': function(value) {
+      this._isVibrateEnabled = value;
+    },
+
+    /**
+     * Play or pause the new audio channel and the active audio channels.
+     *
+     * @param {AudioChannelController} audioChannel The new audio channel.
+     * @return {Promise}
+     */
+    _manageAudioChannels: function(audioChannel) {
+      var promise;
+      var promises = [];
+      if (audioChannel.states.active) {
+        var policies = this.audioChannelPolicy.compete(
+          audioChannel,
+          this._activeAudioChannels,
+          {
+            isNewAudioChannelInBackground:
+              this._isAudioChannelInBackground(audioChannel)
+          }
+        );
+        var policiesForActive = policies.activeAudioChannels;
+        policiesForActive && policiesForActive.forEach((policy) => {
+          promises.push(this._handleAudioChannel(policy));
+        });
+        if (promises.length) {
+          promise = Promise.all(promises)
+            .then(this._handleAudioChannel(policies.newAudioChannel));
+        } else {
+          promise = this._handleAudioChannel(policies.newAudioChannel);
+        }
+      } else {
+        promise = this._resetAudioChannel(audioChannel).then(() => {
+          var promise =
+            this._fadeInFadedOutAudioChannel(audioChannel.channelName);
+          promise && promises.push(promise);
+          if (this._activeAudioChannels.size === 0) {
+            promise = this._resumeBackgroundAudioChannel();
+            promise && promises.push(promise);
+          }
+          if (promises.length) {
+            return Promise.all(promises);
+          }
+        });
+      }
+      return promise;
+    },
+
+    /**
+     * Handle the audio channel.
+     *
+     * @param {Object} policy The policy to handle the audio channel.
+     * @param {AudioChannelController} policy.audioChannel The audio channel.
+     * @param {Boolean} policy.isAllowedToPlay Play or pause it.
+     * @param {Boolean} policy.isNeededToFadeOut Fade out or not.
+     * @param {Boolean} policy.isNeededToVibrate Vibrate or not.
+     * @param {Boolean} policy.isResumedWhenOtherEnds
+     * Resume when other audio channel ends if as true.
+     * @return {Promise}
+     */
+    _handleAudioChannel: function(policy) {
```

Comment:
Discussed with Alive in person.
=> `_handleAudioChannel: function(audioChannel, policy)`